src/components/RightPane/CoursePane/SearchForm/DeptSearchBar/DeptSearchBar.tsx

```diff
@@ -85,6 +106,29 @@ class DeptSearchBar extends PureComponent<DeptSearchBarProps, DeptSearchBarState
         RightPaneStore.updateFormValue('deptValue', setDeptValue.deptValue);
         RightPaneStore.updateFormValue('deptLabel', setDeptValue.deptLabel);
 
+        let stateObj = { url: "url" };
+        const url = new URL(window.location.href)
+        const urlParam = new URLSearchParams(url.search);
+        urlParam.delete('deptLabel');
+        urlParam.delete('deptValue');
+        if (setDeptValue.deptValue != "" && setDeptValue.deptValue != null && 
+            setDeptValue.deptValue != "ALL" && setDeptValue.deptLabel != "" && 
+            setDeptValue.deptLabel != null && setDeptValue.deptLabel != "ALL: Include All Departments")
+            {
+            urlParam.append('deptLabel', setDeptValue.deptLabel);
+            urlParam.append('deptValue', setDeptValue.deptValue);
+            const new_url = `?${urlParam.toString()}`;
+            window.history.replaceState(stateObj, "url", "/" + new_url);
+        }else{
+            if (urlParam.toString() == "" || urlParam.toString() == null){
+                const new_url = `${urlParam.toString()}`;
+                window.history.replaceState(stateObj, "url", "/" + new_url);
+            }else{
+                const new_url = `?${urlParam.toString()}`;
+                window.history.replaceState(stateObj, "url", "/" + new_url);
```

Comment:
It isn't necessarily wrong to save to the window state...but it's kinda hacky if we're using react-router-dom; here's relevant notes in the documentation regarding history and location for the library: https://reactrouter.com/en/main/start/concepts#locations

It may be possible to more cleanly implement this feature if we use react-router-dom's total control over history state.

You don't have to implement this however; it's just something to note if you're working with certain libraries. e.g. this app is *client-side routed*, all  traditional `window.history` calls, navigation, etc. are intercepted and handled by the client side router, not by the browser. The browser history only has entries because the client side router actually added those entries for you.